chore(compiler): Upgrade esy to support new macOS

phated opened the pull request:
@ospencer @marcusroberts can you test this out on your fancy new macOS version?

ospencer commented:
@phated I'm fairly confident it works since I've been using 0.6.12!

ospencer commented:
@phated tested and I'm able to build the compiler. @marcusroberts I don't know if your experience is better, but I have to hop between Rosetta and native running `yarn compiler build` until it eventually succeeds. It's possible that my global esy cache is busted and I need to clear it—maybe then I can do a full Rosetta build just fine.

ospencer commented:
(but that's more an M1 problem than a Monterrey problem, works great on Monterrey!)

marcusroberts commented:
Testing now.  I've been on 0.6.12 for a while too.

I've had to do the initial build under Rosetta, but all yarn compiler clean / yarn compiler build afterwards are done in M1 land with no problem 
for me

marcusroberts commented:
It asked me to choose a version and I went for 0.6.12-beta.2

Apart from that all good.

phated commented:
Esy 0.6.12 is officially out! I've updated this PR and promoted it from draft. I've also reverted the `yarn ... run` stuff because it causes really weird behaviors in esy. I added a note to the contributor docs that you need it installed globally and in your PATH.

phated commented on CONTRIBUTING.md:
```diff
@@ -22,6 +22,8 @@ There are a set of contributor docs in [docs/contributor](https://github.com/gra
 
 ### Compiler
 
+When working with the compiler, you'll want to install [esy][https://esy.sh/docs/en/getting-started.html#install-esy] globally and make it available in your `$PATH`. This will allow you to have LSP services while working on the compiler code.
```
```suggestion
When working with the compiler, you'll want to install [esy](https://esy.sh/docs/en/getting-started.html#install-esy) globally and make it available in your `$PATH`. This will allow you to have LSP services while working on the compiler code.
```

ospencer reviewed (approved):
Looks great!

marcusroberts commented:
Has been working well for me, looks good!

marcusroberts reviewed (approved):
Working well for me!

phated commented:
Going to merge this but pinging @peblair @jozanza and @cician for visibility because it changes the requirement to **need** esy installed globally to have ocamllsp support.
